cmd/submit.go

```diff
@@ -36,23 +36,45 @@ If called with the name of an exercise, it will work out which
 track it is on and submit it. The command will ask for help
 figuring things out if necessary.
 `,
+	Args: cobra.MaximumNArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		usrCfg, err := config.NewUserConfig()
+		// Validate input before doing any other work
+		exercise, err := cmd.Flags().GetString("exercise")
 		if err != nil {
 			return err
 		}
 
-		cliCfg, err := config.NewCLIConfig()
+		trackId, err := cmd.Flags().GetString("track")
```

Comment:
In Go the idiom is to use all caps for ID in variable names, so this should be `trackID`.